### PR TITLE
feat(ui): Custom shape padding

### PR DESCRIPTION
**Pull request by JoseLion**
This PR adds a feature to allow the customization of the padding the shape has around the attached component. Due to these changes, a few tests were also refactored to become less fragile to UI/UX iterations.

### CODE REVIEW

File: package/src/lib/SpotlightTour.provider.tsx [hunk 1]
```diff
@@ -64,6 +64,14 @@ export interface SpotlightTourProviderProps {
    * @default 0.45
    */
   overlayOpacity?: number;
+  /**
+   * Defines the paading of the spot shape based on the wrapped component, so a
```
Comment (alejo0o) [inline package/src/lib/SpotlightTour.provider.tsx:68]:
```suggestion
   * Defines the padding of the spot shape based on the wrapped component, so a
```

File: package/src/lib/SpotlightTour.provider.tsx [hunk 2]
```diff
@@ -76,12 +84,13 @@ export interface SpotlightTourProviderProps {
 export const SpotlightTourProvider = forwardRef<SpotlightTour, SpotlightTourProviderProps>((props, ref) => {
   const {
     children,
+    motion = "bounce",
+    nativeDriver = true,
     onBackdropPress,
     overlayColor = "black",
     overlayOpacity = 0.45,
+    spotPadding = 16,
```
Comment (alejo0o) [inline package/src/lib/SpotlightTour.provider.tsx:92]: Is this the default padding of the wrapped component? cause in the documentation it mentions the ```@default``` is 20

File: package/src/lib/components/tour-overlay/shapes/CircleShape.component.tsx [hunk 3]
```diff
@@ -107,4 +108,4 @@ export const CircleShape = memo<CircleShapeProps>(({ motion, useNativeDriver })
       fill="black"
     />
   );
-}, (prev, next) => prev.motion === next.motion);
+}, deepEqual);
```
Comment (alejo0o) [inline package/src/lib/components/tour-overlay/shapes/CircleShape.component.tsx:111]: 🤯

**Review by alejo0o — APPROVED** [review]
LGTM! 👍 just left minor comments

File: package/src/lib/SpotlightTour.provider.tsx (hunk 2 above)
Comment (JoseLion) [reply package/src/lib/SpotlightTour.provider.tsx:92]: Yes, that's the actual default. I'll change the docs to match this value. Thanks!

**Pull request merged**